agents/meinberg_m1000/meinberg_m1000_agent.py

```diff
@@ -0,0 +1,482 @@
+import os
+from os import environ
+
+import time
+import argparse
+import txaio
+
+from autobahn.twisted.util import sleep as dsleep
+from twisted.internet.defer import inlineCallbacks
+
+from socs.snmp import SNMPTwister
+
+# For logging
+txaio.use_twisted()
+
+on_rtd = os.environ.get('READTHEDOCS') == 'True'
+if not on_rtd:
+    from ocs import ocs_agent, site_config
+
+# Mapping of integer OID values to meaningful strings from M1000 manual
+REFCLOCKSTATE = {0: "refclock is not available",
+                 1: "synchronized",
+                 2: "not synchronized"}
+
+NTPCURRENTSTATE = {0: "not available",
+                   1: "not synchronized",
+                   2: "synchronized"}
+
+SYSPSSTATUS = {0: "notAvailable",
+               1: "down",
+               2: "up"}
+
+ETHPORTLINKSTATE = {0: "notAvailable",
+                    1: "up"}
+
+PTPPORTSTATE = {0: "uninitialized",
+                1: "initializing",
+                2: "faulty",
+                3: "disabled",
+                4: "listening",
+                5: "preMaster",
+                6: "master",
+                7: "passive",
+                8: "uncalibrated",
+                9: "slave"}
+
+
+class MeinbergSNMP:
+    """Meinberg SNMP communicator. Handles communication with and decoding of
+    values from the Meinberg M1000.
+
+    Parameters
+    ----------
+    address : str
+        Address of the M1000.
+    port : int
+        SNMP port to issue GETs to, default to 161.
+
+    Attributes
+    ----------
+    address : str
+        Address of the M1000.
+    port : int
+        SNMP port to issue GETs to.
+    snmp : socs.snmp.SNMPTwister
+        snmp handler from SOCS
+    decoder : dict
+        dict to facilitate the decoding of integer OID values returned from
+        queries to the M1000
+    mib_timings : list
+        list of dicts describing the SNMP OIDs to check, and at which
+        intervals. Each dict contains the keys "oid", "interval", and
+        "lastGet". The corresponding values are of types tuple, integer, and
+        float, respectively. "lastGet" is initialized as None, since no SNMP
+        GET commands have been issued.
+    oid_cache : dict
+        Cache of OID values and corresponding decoded values. Meant to pass to
+        session.data.
+
+    """
+    def __init__(self, address, port=161):
+        self.log = txaio.make_logger()
+        self.address = address
+        self.port = port
+        self.snmp = SNMPTwister(address, port)
+
+        # Decode OID States
+        self.decoder = {'mbgLtNgRefclockState': REFCLOCKSTATE,
+                        'mbgLtNgNtpCurrentState': NTPCURRENTSTATE,
+                        'mbgLtNgSysPsStatus': SYSPSSTATUS,
+                        'mbgLtNgEthPortLinkState': ETHPORTLINKSTATE,
+                        'mbgLtNgPtpPortState': PTPPORTSTATE}
```

Comment:
This map from OID to enumerator could be at global (module) scope.  That would be more useful than only having REFCLOCKSTATE and friends at global scope.  In fact you wouldn't need REFCLOCKSTATE and friends at all.

This is a useful data structure (for interpreting / displaying data downstream, perhaps far in the future), and it doesn't need to be hidden in an instance.

Reply:
Moved the decoder to global scope and got rid of REFCLOCKSTATE and friends, consolidating to one dict.